src/services/blockchain/__tests__/eth-bc-service.test.ts

```diff
@@ -88,3 +86,185 @@ describe('ETH service',  () => {
   })
 
 });
+
+describe('ETH service with mock wallet',  () => {
+  let ethBc: EthereumBlockchainService = null;
+  const provider = {
+    estimateGas: jest.fn(),
+    getBalance: jest.fn(),
+    getBlock: jest.fn(),
+    getGasPrice: jest.fn(),
+    getNetwork: jest.fn(),
+    getTransactionCount: jest.fn(),
+    waitForTransaction: jest.fn(),
```

Comment:
Maybe better to mock `waitForTransaction` as a promise

Reply:
Unless there's a specific reason why I'm inclined to leave it as-is simply because it's working

Reply:
Fair nuff